lib/Backends/NNPI/NNPIResource.cpp

```diff
@@ -544,13 +545,24 @@ bool NNPIResource::updateHostResourceFromTensor(Tensor *t, bool partialTensor) {
 
   // Pad with zeros if needed.
   if (partialData && !partialTensor) {
-    memset(reinterpret_cast<uint8_t *>(hostPtr_) + unpaddedSize, 0,
-           paddedSize - unpaddedSize);
+    if (!requiresNZPadding) {
+      memset(reinterpret_cast<uint8_t *>(hostPtr_) + unpaddedSize, 0,
+             paddedSize - unpaddedSize);
+    } else {
+      LOG_AND_RETURN_IF(
+          ERROR, !downcastInt64,
```

Comment:
You may want to amend this condition. There's nothing preventing padding for any data type and in the future you may want to change the model to already hold indices in 32bit and save the conversion from 64bit.